src/components/modebar/buttons.js

```diff
@@ -0,0 +1,495 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../../plotly');
+var Lib = require('../../lib');
+var Snapshot = require('../../snapshot');
+
+
+var modebarButtons = module.exports = {};
+
+/**
+ * Modebar buttons configuration
+ *
+ * @param {string} name
+ *      name / id of the buttons (for tracking)
+ * @param {string} title
+ *      text that appears while hovering over the button
+ * @param {string} icon
+ *      name of the svg icon associated with the button
```

Comment:
also @etpinard pointed out that for custom icons it would be awesome to be able to use an image - a lot easier for users to generate than svg. once `button.icon` is an object, that would be pretty easy to do (in another PR). We could also make an option for text as the icon.